internal/gatewayapi/translator.go

```diff
@@ -156,7 +157,17 @@ func newTranslateResult(gateways []*GatewayContext,
 
 func (t *Translator) Translate(resources *resource.Resources) (*TranslateResult, error) {
 	// Get Gateways belonging to our GatewayClass.
-	gateways := t.GetRelevantGateways(resources)
+	allGateways := t.GetRelevantGateways(resources)
+
+	// Filter out Gateways that are marked as not accepted by the controller.
```

Comment:
this means a bad config on EnvoyProxy can take down all data plane xds config ?

Reply:
also, thoughts on changing the api of `GetRelevantGateways` to `acceptedGateways, failedGateways := GetRelevantGateways(res)` ?

Reply:
> this means a bad config on EnvoyProxy can take down all data plane xds config ?

Yes, the current behavior is that the Gateway will be still accepted and the envoy fleet will be deployed.
We can keep the current behavior, but there is no other condition type to reflect a "partial error" status. The Gateway can only be either accepted or not.

Reply:
Or we could set `Accepted` as false but still provision the Envoy fleet and server the xDS.

Reply:
sounds good, lets document this in the breaking change that any issues with `EnvoyProxy` reference in a gateway resulting in a `Accepted=False` will result in no xDS being pushed to the fleet